akka-remote/src/test/scala/akka/remote/artery/OutboundIdleShutdownSpec.scala

```diff
@@ -8,59 +8,147 @@ import akka.remote.RARP
 import akka.testkit.ImplicitSender
 import akka.testkit.TestActors
 import akka.testkit.TestProbe
+import org.scalatest.concurrent.Eventually
+import scala.concurrent.duration._
 
 class OutboundIdleShutdownSpec extends ArteryMultiNodeSpec("""
   akka.loglevel=DEBUG
-  akka.remote.artery.advanced.stop-idle-outbound-after = 3 s
-  """) with ImplicitSender {
+  akka.remote.artery.advanced.stop-idle-outbound-after = 1 s
+  """) with ImplicitSender with Eventually {
 
-  "Idle outbound associations" should {
-
-    "be cleaned up" in {
+  "Outbound lanes" should {
 
+    "eliminate an association when all of them are idle" in {
       val remoteSystem = newRemoteSystem()
       remoteSystem.actorOf(TestActors.echoActorProps, "echo")
       val remoteAddress = RARP(remoteSystem).provider.getDefaultAddress
 
-      def sel = system.actorSelection(RootActorPath(remoteAddress) / "user" / "echo")
+      def remoteEcho = system.actorSelection(RootActorPath(remoteAddress) / "user" / "echo")
       val localProbe = new TestProbe(localSystem)
-      sel.tell("ping", localProbe.ref)
+      remoteEcho.tell("ping", localProbe.ref)
       localProbe.expectMsg("ping")
 
-      val accociation = RARP(system).provider.transport.asInstanceOf[ArteryTransport].association(remoteAddress)
-      accociation.isStreamActive(Association.ControlQueueIndex) should ===(true)
-      accociation.isStreamActive(Association.OrdinaryQueueIndex) should ===(true)
+      val artery = RARP(system).provider.transport.asInstanceOf[ArteryTransport]
+      val association = artery.association(remoteAddress)
+      withClue("When initiating a connection, both the control - and ordinary lanes are opened (regardless of which one was used)") {
+        // TODO: Is this still desirable? Should lanes perhaps be opened opportunistically always?
```

Comment:
Initially the streams are started eagerly, but after idle stop they are started on demand (on first message).

The control stream will always be activated though, because the handshake response messages are sent over control stream.

Reply:
Right, that last bit is something I missed before. Removed the comment